asf_heat_pump_suitability/research/exploratory/sample_buildings_for_labelling.py

```diff
@@ -0,0 +1,401 @@
+    .alias("n_flats_grouped"),
+    # Add labeller
+    pl.Series("labeller", ["reindeer"] * already_labelled_df.height),
+)
+
+# Get buildings containing flats
+buildings_containing_flats = sampling_areas_df.filter(pl.col("property_type_flat"))[
+    "building_id"
+].unique(maintain_order=True)
+
+# Sample buildings containing flats but only if they haven't been labelled
+sample_from_df = (
+    sampling_areas_df.filter(
+        pl.col("building_id").is_in(buildings_containing_flats),
+        ~pl.col("building_id").is_in(already_labelled),
+    )
+    .group_by("building_id")
+    .agg(
+        pl.col("property_type_flat").sum().alias("n_flats"),
+        pl.col("UPRN").count().alias("n_total"),
+        pl.col("LAD23NM").first().alias("la_name"),
+        # Sample from any buildings with >1 flat
+    )
+    .filter(pl.col("n_flats") > 1)
+    .with_columns(
+        # Group number of flats
+        pl.when(pl.col("n_flats").is_between(2, 6, closed="both"))
+        .then(pl.lit("2-6_flats"))
+        .when(pl.col("n_flats").is_between(7, 15, closed="both"))
+        .then(pl.lit("7-15_flats"))
+        .when(pl.col("n_flats") > 15)
+        .then(pl.lit("16+_flats"))
+        .otherwise(None)
+        .alias("n_flats_grouped")
+    )
+)
+
+samples = []
+
+n_flats_groups = ["2-6_flats", "7-15_flats", "16+_flats"]
+
+# Take samples for each Local Authority for each group of flat counts to create sample of 700 buildings
+for la, df in sample_from_df.sort(by="la_name").group_by(
+    "la_name", maintain_order=True
+):
+    # Get the number of samples per LA
+    n = n_samples[la[0]]
+    # Take one third of samples from each group of flat counts
+    for n_flats in n_flats_groups:
+        _n = np.ceil(n / len(n_flats_groups))
+        samples.append(
+            df.filter(pl.col("n_flats_grouped") == n_flats)
+            .sort(by="building_id")
+            .sample(n=_n, seed=seed)
+        )
+
+sample_3_df = pl.concat(samples)
+
+# %%
+# ASSIGN LABELLERS TO SAMPLE
+# Get number of samples per labeller
+n_samples_per_labeller = int(np.ceil(sample_3_df.height / len(labellers)))
+# Create list of labellers corresponding to total sample length
+labeller_assignments = labellers * n_samples_per_labeller
+
+# Shuffle the list for random assignment
+random.seed(seed)
+random.shuffle(labeller_assignments)
+
+# Assign a labeller to each row
+labeller_assignments = labeller_assignments[: sample_3_df.height]
+sample_3_df = sample_3_df.with_columns(pl.Series("labeller", labeller_assignments))
+
+second_samples = []
+final_samples = {}
+
+# Create the final subset for each labeller
+for labeller in labellers:
+    # Get the number of primary samples selected for the labeller
+    n_samples = sample_3_df.filter(pl.col("labeller") == labeller).height
+
+    # Calculate number of additional samples so that some buildings get 2 labels from different people
+    total_sample = n_samples_per_labeller + 30
+    additional_sample = total_sample - n_samples
```

Comment:
Apologies if this is obvious, but I'm getting a little bit confused with the + 30 here. Are we adding 30 extra samples for each labeller? If so, why are we computing the additional_sample?